srv/modules/runners/validate.py

```diff
@@ -589,13 +596,16 @@ def openattic(self):
         """
         local = salt.client.LocalClient()
         for node in self.data.keys():
-            if ('roles' in self.data[node] and
-                'openattic' in self.data[node]['roles']):
-                # Would use file.contains if it supported '='
-                result = local.cmd(node, 'file.contains_regex', [ '/etc/ceph/ceph.conf', 'port\=80'], expr_form="glob")
-                if result[node]:
-                    msg = "rgw port conflicts with openATTIC on {} - check ceph.conf".format(node)
-                    self.errors.setdefault('openattic', []).append(msg)
+            if ('roles' in self.data[node]):
+                log.debug("Considering node {} with roles {}".format(node, self.data[node]['roles']))
+                if ('openattic' in self.data[node]['roles'] and
+                    'rgw' in self.data[node]['roles']):
+                    log.info("Detected colocated rgw and openattic on node {}".format(node))
+                    # Would use file.contains if it supported '='
+                    result = local.cmd(node, 'file.contains_regex', [ '/etc/ceph/ceph.conf', 'port\=80'], expr_form="glob")
```

Comment:
this wouldn't match anything yet since validate will be run pre-stage3 and configuration gets deployed during stage 3

Reply:
@theanalyst Not true. IIRC this code is triggered by `validate.deploy` which is run as part of Stage 4.

Reply:
I see